### ISSUE TITLE
Installing for Stupid people

### ISSUE THREAD

**Issue by Megasherru** [zjp-shadow/CharacterGen#19]
Im sure im doing something wrong. 
Im not sure. I dont know if its assumed that something else would have been installed already. 

One thing, I have Python 3.10 and not 3.9 (unsure if this is the issue)
But here is what I did

1. Create a folder: I then do. CMD through the address bar then
Git Clone: https://github.com/zjp-shadow/CharacterGen.git

2, pip install -r requirements.txt

3. Next I do: 
huggingface-cli download --resume-download zjpshadow/CharacterGen --include 2D_Stage/* --local-dir .
huggingface-cli download --resume-download zjpshadow/CharacterGen --include 3D_Stage/* --local-dir .

4. Then this is the part I am having issues with. I now go to the 2D folder, and I run the Webui. but it doesnt actually run. 
The thing flashes and closes. 

Again, im thinking im being stupid somewhere. or it was meant to be run through something else. but not sure what that would be.
What am I doing wrong? help for a complete layman noob would be awesome!

**Comment by JimikoSK** [comment]
I don't think it works unless you explicitly use 3.9, I was using 3.9.1 first, but it did not work.

**Comment by N3X15** [comment]
Yeah, needs to be 3.9.  Some dependencies do not like anything higher:

![image](https://github.com/user-attachments/assets/baf56943-f221-46e1-92bb-5db12579c38a)

**Comment by jkerman2000** [comment]
> Yeah, needs to be 3.9. Some dependencies do not like anything higher:
> 
> ![image](https://private-user-images.githubusercontent.com/110073/354226955-baf56943-f221-46e1-92bb-5db12579c38a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAxNzY0NjcsIm5iZiI6MTczMDE3NjE2NywicGF0aCI6Ii8xMTAwNzMvMzU0MjI2OTU1LWJhZjU2OTQzLWYyMjEtNDZlMS05MmJiLTVkYjEyNTc5YzM4YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyOVQwNDI5MjdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04YmE3Y2UzNWY4ZDZkZmFiYTY0Mzk2M2M3OGRhNTcyOTVhZTg1NzdhZDYwMTgyNmMxNmZlOTAxZWY5Njg4YzI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.1ExPAkeTi5gDKwvPTirK62Q7wbx3m-AtdIFGHg4uuF8)
> 
> (I am using a non-standard install using poetry so I can package it up for some dumb people I know.)

Any chance I could get my hands on a copy of the package? My python installs are mess and nothing installs correctly anymore.

**Comment by Jillinger** [comment]
Since I don't have an issue with the actual program, but am a bit clueless on a couple things, I thought this might be an appropriate place to ask.

1. I was able to use `python webui.py` in both 2d and 3d Stage directories, and I got pytorch_model.bin etc. I changed my CUDA version, and Python version, and used `huggingface-cli download --resume-download zjpshadow/CharacterGen --include 2D_Stage/* --local-dir . `which got pytorch_model.bin.metadata. Should the final result be .metadata extension or .bin extension?
2. Which directory should I download the UniRig model to - `hf download VAST-AI/UniRig`?

**Comment by Jillinger** [comment]
> Im sure im doing something wrong. Im not sure. I dont know if its assumed that something else would have been installed already.
> 
> One thing, I have Python 3.10 and not 3.9 (unsure if this is the issue) But here is what I did
> 
> 1. Create a folder: I then do. CMD through the address bar then
>    Git Clone: https://github.com/zjp-shadow/CharacterGen.git
> 
> 2, pip install -r requirements.txt
> 
> 3. Next I do:
>    huggingface-cli download --resume-download zjpshadow/CharacterGen --include 2D_Stage/* --local-dir .
>    huggingface-cli download --resume-download zjpshadow/CharacterGen --include 3D_Stage/* --local-dir .
> 4. Then this is the part I am having issues with. I now go to the 2D folder, and I run the Webui. but it doesnt actually run.
>    The thing flashes and closes.
> 
> Again, im thinking im being stupid somewhere. or it was meant to be run through something else. but not sure what that would be. What am I doing wrong? help for a complete layman noob would be awesome!

I'm not 100% sure, but it could be you might be missing some python packages, or you could try Python 3.13.
I was able to use `python webui.py` with that version, and get the files.
I got an error when using Python 3.12, but didn't investigate the cause. Instead, I ran `huggingface-cli download --resume-download zjpshadow/CharacterGen --include 2D_Stage/* --local-dir .`, and the downloads completed.
Do I need to run `python webui.py` now? Is that why the file extensions are metadata?
Looking at the file size, it looks like that is the case.
